apps/wakunode2/app.nim

```diff
@@ -335,83 +470,11 @@ proc initNode(conf: WakuNodeConf,
 
     dnsResolver = DnsResolver.new(nameServers)
 
-  let
-    nodekey = if conf.nodekey.isSome():
-                conf.nodekey.get()
-              else:
-                let nodekeyRes = crypto.PrivateKey.random(Secp256k1, rng[])
-                if nodekeyRes.isErr():
-                  return err("failed to generate nodekey: " & $nodekeyRes.error)
-                nodekeyRes.get()
-
-
-  ## `udpPort` is only supplied to satisfy underlying APIs but is not
-  ## actually a supported transport for libp2p traffic.
-  let udpPort = conf.tcpPort
-  let natRes = setupNat(conf.nat, clientId,
-                        Port(uint16(conf.tcpPort) + conf.portsShift),
-                        Port(uint16(udpPort) + conf.portsShift))
-  if natRes.isErr():
-    return err("failed to setup NAT: " & $natRes.error)
-
-  let (extIp, extTcpPort, _) = natRes.get()
-
-  let
-    dns4DomainName = if conf.dns4DomainName != "": some(conf.dns4DomainName)
-                      else: none(string)
-
-    discv5UdpPort = if conf.discv5Discovery: some(Port(uint16(conf.discv5UdpPort) + conf.portsShift))
-                    else: none(Port)
-
-    ## TODO: the NAT setup assumes a manual port mapping configuration if extIp config is set. This probably
-    ## implies adding manual config item for extPort as well. The following heuristic assumes that, in absence of manual
-    ## config, the external port is the same as the bind port.
-    extPort = if (extIp.isSome() or dns4DomainName.isSome()) and extTcpPort.isNone():
-                some(Port(uint16(conf.tcpPort) + conf.portsShift))
-              else:
-                extTcpPort
-    extMultiAddrs = if (conf.extMultiAddrs.len > 0):
-                      let extMultiAddrsValidationRes = validateExtMultiAddrs(conf.extMultiAddrs)
-                      if extMultiAddrsValidationRes.isErr():
-                        return err("invalid external multiaddress: " & extMultiAddrsValidationRes.error)
-                      else:
-                        extMultiAddrsValidationRes.get()
-                    else:
-                      @[]
-
-    wakuFlags = CapabilitiesBitfield.init(
-        lightpush = conf.lightpush,
-        filter = conf.filter,
-        store = conf.store,
-        relay = conf.relay
-      )
-
   var node: WakuNode
 
   let pStorage = if peerStore.isNone(): nil
                  else: peerStore.get()
 
-  let rng = crypto.newRng()
```

Comment:
Mmm. Wait, was this rng generation redundant before as it's also an argument for this function?

Reply:
Yes that's what I concluded.